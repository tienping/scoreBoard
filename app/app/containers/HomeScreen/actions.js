/*
 *
 * HomeScreen actions
 *
 */

import {
    DEFAULT_ACTION,
    FETCH_STORE,
    FETCH_STORE_FAILED,
    FETCH_STORE_SUCCESS,
    ATTENDANCE_REQUEST,
    ATTENDANCE_SUCCESS,
    ATTENDANCE_FAILED,
    FETCH_RECENTLY_VIEWED,
    FETCH_RECENTLY_VIEWED_SUCCESS,
    FETCH_RECENTLY_VIEWED_FAILED,
    CLEAR_RECENTLY_VIEWED,
    CLEAR_RECENTLY_VIEWED_SUCCESS,
    CLEAR_RECENTLY_VIEWED_FAILED,
    FETCH_RECOMMENDED_PRODUCT,
    FETCH_RECOMMENDED_PRODUCT_SUCCESS,
    FETCH_RECOMMENDED_PRODUCT_FAILED,
} from './constants';

export function defaultAction() {
    return {
        type: DEFAULT_ACTION,
    };
}
export function fetchStore() {
    return {
        type: FETCH_STORE,
    };
}

export function fetchStoreSuccess(response) {
    return {
        type: FETCH_STORE_SUCCESS,
        payload: response,
    };
}

export function fetchStoreFailed(response) {
    return {
        type: FETCH_STORE_FAILED,
        payload: response,
    };
}

export function attendanceRequest() {
    return {
        type: ATTENDANCE_REQUEST,
    };
}

export function attendanceSuccess(response, token) {
    return {
        type: ATTENDANCE_SUCCESS,
        payload: {
            response,
            token,
        },
    };
}

export function attendanceFailed(response, token) {
    return {
        type: ATTENDANCE_FAILED,
        payload: {
            response,
            token,
        },
    };
}

export function fetchRecentlyViewed() {
    return {
        type: FETCH_RECENTLY_VIEWED,
    };
}

export function fetchRecentlyViewedSuccess(response) {
    return {
        type: FETCH_RECENTLY_VIEWED_SUCCESS,
        payload: response,
    };
}

export function fetchRecentlyViewedFailed(response) {
    return {
        type: FETCH_RECENTLY_VIEWED_FAILED,
        payload: response,
    };
}

export function clearRecentlyViewed() {
    return {
        type: CLEAR_RECENTLY_VIEWED,
    };
}

export function clearRecentlyViewedSuccess(response) {
    return {
        type: CLEAR_RECENTLY_VIEWED_SUCCESS,
        payload: response,
    };
}

export function clearRecentlyViewedFailed(response) {
    return {
        type: CLEAR_RECENTLY_VIEWED_FAILED,
        payload: response,
    };
}

export function fetchRecommendedProduct(response) {
    return {
        type: FETCH_RECOMMENDED_PRODUCT,
        payload: response,
    };
}
export function fetchRecommendedProductSuccess(response, token) {
    return {
        type: FETCH_RECOMMENDED_PRODUCT_SUCCESS,
        payload: response,
        token,
    };
}

export function fetchRecommendedProductFailed(response) {
    return {
        type: FETCH_RECOMMENDED_PRODUCT_FAILED,
        payload: response,
    };
}
