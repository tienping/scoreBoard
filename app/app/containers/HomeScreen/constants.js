/*
 *
 * HomeScreen constants
 *
 */

export const DEFAULT_ACTION = 'app/HomeScreen/DEFAULT_ACTION';
export const FETCH_STORE = 'app/HomeScreen/FETCH_STORE';
export const FETCH_STORE_SUCCESS = 'app/HomeScreen/FETCH_STORE_SUCCESS';
export const FETCH_STORE_FAILED = 'app/HomeScreen/FETCH_STORE_FAILED';

export const FETCH_TWOH_SUCCESS = 'app/Twoh/FETCH_TWOH_SUCCESS';
export const FETCH_TWOH_FAILED = 'app/Twoh/FETCH_TWOH_FAILED';

export const ATTENDANCE_REQUEST = 'app/HomeScreen/ATTENDANCE_REQUEST';
export const ATTENDANCE_SUCCESS = 'app/HomeScreen/ATTENDANCE_SUCCESS';
export const ATTENDANCE_FAILED = 'app/HomeScreen/ATTENDANCE_FAILED';

export const FETCH_RECOMMENDED_PRODUCT = 'app/HomeScreen/FETCH_RECOMMENDED_PRODUCT';
export const FETCH_RECOMMENDED_PRODUCT_SUCCESS = 'app/HomeScreen/FETCH_RECOMMENDED_PRODUCT_SUCCESS';
export const FETCH_RECOMMENDED_PRODUCT_FAILED = 'app/HomeScreen/FETCH_RECOMMENDED_PRODUCT_FAILED';

// Recently Viewed Constant
export const FETCH_RECENTLY_VIEWED = 'app/HomeScreen/FETCH_RECENTLY_VIEWED';
export const FETCH_RECENTLY_VIEWED_SUCCESS = 'app/HomeScreen/FETCH_RECENTLY_VIEWED_SUCCESS';
export const FETCH_RECENTLY_VIEWED_FAILED = 'app/HomeScreen/FETCH_RECENTLY_VIEWED_FAILED';
export const CLEAR_RECENTLY_VIEWED = 'app/HomeScreen/CLEAR_RECENTLY_VIEWED';
export const CLEAR_RECENTLY_VIEWED_SUCCESS = 'app/HomeScreen/CLEAR_RECENTLY_VIEWED_SUCCESS';
export const CLEAR_RECENTLY_VIEWED_FAILED = 'app/HomeScreen/CLEAR_RECENTLY_VIEWED_FAILED';

