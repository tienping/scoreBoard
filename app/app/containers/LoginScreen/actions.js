/*
 *
 * LoginScreen actions
 *
 */

import {
    AUTH_LOGIN,
    AUTH_LOGIN_SUCCESS,
    AUTH_LOGIN_FAILED,
    AUTH_LOGOUT,
    AUTH_FBLOGIN,
    FORGET_PASSWORD_FAILED,
    FORGET_PASSWORD_REQUEST,
    FORGET_PASSWORD_SUCCESS,
    RESET_MESSAGE_REQUEST,
} from './constants';

export const login = (username, password) => ({
    type: AUTH_LOGIN,
    payload: {
        username,
        password,
    },
});

export const loginFB = (id, token, email) => ({
    type: AUTH_FBLOGIN,
    payload: {
        id,
        token,
        email,
    },
});

export const loginSuccess = (key) => ({
    type: AUTH_LOGIN_SUCCESS,
    payload: {
        token: key,
    },
});

export const loginFail = (response) => ({
    type: AUTH_LOGIN_FAILED,
    payload: response,
});

export const logout = () => ({
    type: AUTH_LOGOUT,
});

export const forgetRequest = (email) => ({
    type: FORGET_PASSWORD_REQUEST,
    payload: {
        email,
    },
});

export const forgetSuccess = (response) => ({
    type: FORGET_PASSWORD_SUCCESS,
    payload: response,
});

export const forgetFailed = (response) => ({
    type: FORGET_PASSWORD_FAILED,
    payload: response,
});

export function resetMessage() {
    return {
        type: RESET_MESSAGE_REQUEST,
    };
}
