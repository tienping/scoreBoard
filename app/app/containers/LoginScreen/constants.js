/*
 *
 * LoginScreen constants
 *
 */

export const AUTH_FBLOGIN = 'app/LoginScreen/AUTH_FBLOGIN';
export const AUTH_LOGIN = 'app/LoginScreen/AUTH_LOGIN';
export const AUTH_LOGIN_SUCCESS = 'app/LoginScreen/AUTH_LOGIN_SUCCESS';
export const AUTH_LOGIN_FAILED = 'app/LoginScreen/AUTH_LOGIN_FAILED';

export const AUTH_LOGOUT = 'app/LoginScreen/AUTH_LOGOUT';

export const FORGET_PASSWORD_REQUEST = 'app/LoginScreen/FORGET_PASSWORD_REQUEST';
export const FORGET_PASSWORD_SUCCESS = 'app/LoginScreen/FORGET_PASSWORD_SUCCESS';
export const FORGET_PASSWORD_FAILED = 'app/LoginScreen/FORGET_PASSWORD_FAILED';
export const RESET_MESSAGE_REQUEST = 'app/LoginScreen/RESET_MESSAGE_REQUEST';
