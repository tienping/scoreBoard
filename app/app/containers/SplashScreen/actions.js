/*
 *
 * SplashScreen actions
 *
 */

import {
    FETCH_GET_COMMON_DATA, FETCH_GET_COMMON_DATA_FAILED, FETCH_GET_COMMON_DATA_SUCCESS,
} from './constants';

export function fetchGetCommonData() {
    return {
        type: FETCH_GET_COMMON_DATA,
    };
}

export function fetchGetCommonDataSuccess(response) {
    return {
        type: FETCH_GET_COMMON_DATA_SUCCESS,
        payload: response,
    };
}

export function fetchGetCommonDataFailed(response) {
    return {
        type: FETCH_GET_COMMON_DATA_FAILED,
        payload: response,
    };
}
