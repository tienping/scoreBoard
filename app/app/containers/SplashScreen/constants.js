/*
 *
 * SplashScreen constants
 *
 */

export const FETCH_GET_COMMON_DATA = 'app/SplashScreen/FETCH_GET_COMMON_DATA';
export const FETCH_GET_COMMON_DATA_SUCCESS = 'app/SplashScreen/FETCH_GET_COMMON_DATA_SUCCESS';
export const FETCH_GET_COMMON_DATA_FAILED = 'app/SplashScreen/FETCH_GET_COMMON_DATA_FAILED';
