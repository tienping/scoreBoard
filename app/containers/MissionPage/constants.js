/*
 *
 * MissionPage constants
 *
 */

export const ADD_ENTRY = 'app/MissionPage/ADD_ENTRY';
