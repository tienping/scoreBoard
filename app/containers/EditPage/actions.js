/*
 *
 * EditPage actions
 *
 */

import { ADD_ENTRY } from './constants';

export function addEntry(data) {
  return {
    type: ADD_ENTRY,
    data,
  };
}
