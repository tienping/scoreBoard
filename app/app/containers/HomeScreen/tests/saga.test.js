/**
 * Test  sagas
 */

/* eslint-disable redux-saga/yield-effects */
import { takeLatest, put } from 'redux-saga/effects';
import storeSaga, { getStoreData, getTWOHData } from '../saga';
import { FETCH_STORE, FETCH_TWOH } from '../constants';
import { fetchStoreFailed, fetchStoreSuccess, fetchTWOHFailed, fetchTWOHSuccess } from '../actions';

const responseSuccess = { items: [] };
const responseFailed = { success: false };

describe('getStoreData Saga', () => {
    let generator;
    beforeEach(() => {
        generator = getStoreData();
        const selectDescriptor = generator.next().value;
        expect(selectDescriptor).toMatchSnapshot();
    });

    it('should dispatch the fetchStoreSuccess if the call is successful', () => {
        const putDescriptor = generator.next(responseSuccess).value;
        expect(putDescriptor).toEqual(put(fetchStoreSuccess(responseSuccess)));
    });

    it('should dispatch the fetchStoreFailed if API returned error', () => {
        const putDescriptor = generator.next(responseFailed).value;
        expect(putDescriptor).toEqual(put(fetchStoreFailed(responseFailed)));
    });
});

describe('getTWOHData Saga', () => {
    let generator;
    beforeEach(() => {
        generator = getTWOHData();
        const selectDescriptor = generator.next().value;
        expect(selectDescriptor).toMatchSnapshot();
    });

    it('should dispatch the fetchTWOHSuccess if the call is successful', () => {
        const putDescriptor = generator.next(responseSuccess).value;
        expect(putDescriptor).toEqual(put(fetchTWOHSuccess(responseSuccess)));
    });

    it('should dispatch the fetchTWOHFailed if API returned error', () => {
        const putDescriptor = generator.next(responseFailed).value;
        expect(putDescriptor).toEqual(put(fetchTWOHFailed(responseFailed)));
    });
});

describe('storeSaga', () => {
    const generator = storeSaga();

    it('should start watching FETCH_STORE action', () => {
        const latestDescriptor = generator.next().value;
        expect(latestDescriptor).toEqual(takeLatest(FETCH_STORE, getStoreData));
    });
    it('should start watching FETCH_TWOH action', () => {
        const latestDescriptor = generator.next().value;
        expect(latestDescriptor).toEqual(takeLatest(FETCH_TWOH, getTWOHData));
    });
});
