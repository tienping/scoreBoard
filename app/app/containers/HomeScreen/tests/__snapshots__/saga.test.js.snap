// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getStoreData Saga should dispatch the fetchStoreFailed if API returned error 1`] = `
Object {
  "@@redux-saga/IO": true,
  "CALL": Object {
    "args": Array [
      "https://devapi.hermo.my/app/store",
    ],
    "context": null,
    "fn": [Function],
  },
}
`;

exports[`getStoreData Saga should dispatch the fetchStoreSuccess if the call is successful 1`] = `
Object {
  "@@redux-saga/IO": true,
  "CALL": Object {
    "args": Array [
      "https://devapi.hermo.my/app/store",
    ],
    "context": null,
    "fn": [Function],
  },
}
`;

exports[`getTWOHData Saga should dispatch the fetchTWOHFailed if API returned error 1`] = `
Object {
  "@@redux-saga/IO": true,
  "CALL": Object {
    "args": Array [
      "https://devapi.hermo.my/app/home/feature-banner",
    ],
    "context": null,
    "fn": [Function],
  },
}
`;

exports[`getTWOHData Saga should dispatch the fetchTWOHSuccess if the call is successful 1`] = `
Object {
  "@@redux-saga/IO": true,
  "CALL": Object {
    "args": Array [
      "https://devapi.hermo.my/app/home/feature-banner",
    ],
    "context": null,
    "fn": [Function],
  },
}
`;
